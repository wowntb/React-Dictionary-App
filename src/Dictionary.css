html {
  background-color: #6a8dff;
}

h1 {
  color: white;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  font-size: 48pt;
  text-align: center;
  border-radius: 5px;
  padding: 10px;
  text-shadow: 0px 2px 2px black;
}

p {
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
}

footer { 
  position: absolute;
  padding: 5px;
  margin: auto;
  bottom: 0px;
}

.section {
  background-color: white;
  color: #346ea5;
  font-size: 16pt;
  padding: 10px;
  width: 60%;
  border-radius: 10px;
  margin-top: 20px;
  margin-left: auto;
  margin-right: auto;
}

