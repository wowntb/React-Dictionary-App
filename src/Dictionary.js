/* Dear code reviewer, I was meant to use the Merriam Webster API but their automated system 
would not allow me to sign up. I contacted a mentor and they allowed me to use an alternative: WordsAPI. */
// This is a React dictionary app. The user can enter a word and receive the definition of it.
import { useState, useEffect } from "react";
import "./Dictionary.css";
// I have avoided hardcoding my API key and used .env to store it.
const apiKey = process.env.REACT_APP_WORDS_API_KEY;

function Dictionary() {
  const [searchTerm, setSearchTerm] = useState("");
  const [data, setData] = useState(null);

  // useEffect is being utilised so that when the state of searchTerm is altered the program fetches data from the API again.
  useEffect(() => {
    if (searchTerm !== "") {
      const apiUrl = `https://wordsapiv1.p.rapidapi.com/words/${searchTerm}`;

      fetch(apiUrl, {
        method: "GET",
        headers: {
          "X-RapidAPI-Key": apiKey,
          "X-RapidAPI-Host": "wordsapiv1.p.rapidapi.com",
        },
      })
        // The response is parsed to JSON.
        .then((response) => response.json())
        // The JSON is saved in state.
        .then((result) => setData(result))
        // This is a line for catching an with the fetch.
        .catch((error) => console.error("Error fetching data:", error));
    }
  }, [searchTerm]);

  const handleSearch = () => {
    setSearchTerm(document.getElementById("searchInput").value);
  };

  return (
    <>
      <h1>React Dictionary App</h1>
      <div>
        <p style={{ textAlign: "center" }}>
          Would you like the definition of a word? Simply type the word in the
          search bar below.
        </p>
        <div style={{ textAlign: "center" }}>
          <input
            type="text"
            id="searchInput"
            placeholder="Enter a word"
            style={{ width: "400px" }}
          />
          <button onClick={handleSearch}>Search</button>
        </div>
      </div>
      {data && (
        <div className="section">
          <h2>{searchTerm}</h2>
          <p>
            Definition: <em>{data.results && data.results[0].definition}.</em>
          </p>
        </div>
      )}

      <footer>Coded by Nathenale Bedane</footer>
    </>
  );
}

export default Dictionary;
