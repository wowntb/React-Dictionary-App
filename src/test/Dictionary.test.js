import React from "react";
import Dictionary from "../Dictionary";
import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<Dictionary></Dictionary>).toJSON();
  expect(tree).toMatchSnapshot();
});
