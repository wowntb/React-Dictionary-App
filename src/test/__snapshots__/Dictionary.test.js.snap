// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1>
    React Dictionary App
  </h1>,
  <div>
    <p
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Would you like the definition of a word? Simply type the word in the search bar below.
    </p>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <input
        id="searchInput"
        placeholder="Enter a word"
        style={
          Object {
            "width": "400px",
          }
        }
        type="text"
      />
      <button
        onClick={[Function]}
      >
        Search
      </button>
    </div>
  </div>,
  <footer>
    Coded by Nathenale Bedane
  </footer>,
]
`;
